packages/cozy-harvest-lib/test/components/FlowProvider.spec.js

```diff
@@ -57,59 +55,65 @@ describe('FlowProvider', () => {
     jest.resetAllMocks()
   })
 
-  it('should render', () => {
-    const component = shallow(
-      <FlowProvider {...props}>{() => <Child />}</FlowProvider>
-    ).getElement()
-    expect(component).toMatchSnapshot()
+  it('should support a flow with a SUCCESS', () => {
+    const children = jest.fn()
+    const wrapper = shallow(<FlowProvider {...props}>{children}</FlowProvider>)
+
+    const flow = wrapper.instance().flow
+    expect(children.mock.calls[0][0].flow.getState().status).toEqual('IDLE')
+
+    flow.triggerEvent(TWO_FA_REQUEST_EVENT)
+    wrapper.update()
+    expect(wrapper.find(TwoFAModal).length).toBe(1)
+
+    flow.triggerEvent(SUCCESS_EVENT)
+    wrapper.update()
+    expect(wrapper.find(TwoFAModal).length).toBe(0)
+    expect(children.mock.calls[2][0].flow.getState().status).toEqual('SUCCESS')
+  })
+
+  it('should support a flow with a LOGIN_SUCCESS', () => {
+    const children = jest.fn()
+    const wrapper = shallow(<FlowProvider {...props}>{children}</FlowProvider>)
+
+    const flow = wrapper.instance().flow
+
+    flow.triggerEvent(TWO_FA_REQUEST_EVENT)
+    wrapper.update()
+    expect(wrapper.find(TwoFAModal).length).toBe(1)
+
+    flow.triggerEvent(LOGIN_SUCCESS_EVENT)
+    wrapper.update()
+    expect(wrapper.find(TwoFAModal).length).toBe(0)
+    expect(children.mock.calls[2][0].flow.getState().status).toEqual(
+      'LOGIN_SUCCESS'
+    )
```

Comment:
Since the flow instance does not change between calls, wouldn't it be simpler to simply do:

```
    expect(flow.getState().status).toEqual(
      'LOGIN_SUCCESS'
    )
```

Reply:
It would be simpler, but the current approach really tests what the child component of `FlowProvider` receives, without relying on the internal implementation of it :)